src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -392,57 +384,64 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
-                            {
-                                CoreModel.NonFungibleIdType.String => NonFungibleIdType.String,
-                                CoreModel.NonFungibleIdType.Integer => NonFungibleIdType.Integer,
-                                CoreModel.NonFungibleIdType.Bytes => NonFungibleIdType.Bytes,
-                                CoreModel.NonFungibleIdType.UUID => NonFungibleIdType.UUID,
-                                _ => throw new ArgumentOutOfRangeException(nameof(e.NonFungibleIdType), e.NonFungibleIdType, "Unexpected value of NonFungibleIdType"),
-                            });
-                        }
+                        referencedEntities
+                            .GetOrAdd(dataTable.EntityIdHex, _ => new ReferencedEntity(dataTable.EntityIdHex, dataTable.EntityType, stateVersion))
+                            .PostResolveConfigure((KeyValueStoreEntity e) => e.StoreOfNonFungibleResourceEntityId = referencedEntity.DatabaseId);
                     }
 
                     if (substateData is CoreModel.VaultInfoSubstate vaultInfo)
                     {
                         referencedEntity.PostResolveConfigure((VaultEntity e) =>
                         {
-                            e.ResourceManagerEntityId = referencedEntities.GetByGlobal((GlobalAddress)vaultInfo.ResourceAddress).DatabaseId;
+                            e.ResourceEntityId = referencedEntities.GetByGlobal((GlobalAddress)vaultInfo.ResourceAddress).DatabaseId;
                         });
                     }
 
                     if (substateData is CoreModel.TypeInfoSubstate typeInfoSubstate)
                     {
-                        referencedEntity.PostResolveConfigure((ComponentEntity e) =>
+                        switch (typeInfoSubstate.Details)
                         {
-                            e.PackageId = referencedEntities.GetByGlobal((GlobalAddress)typeInfoSubstate.PackageAddress).DatabaseId;
-                            e.BlueprintName = typeInfoSubstate.BlueprintName;
-                        });
+                            case CoreModel.KeyValueStoreTypeInfoDetails _:
+                                // TODO schema not supported yet
+                                break;
+                            case CoreModel.ObjectTypeInfoDetails objectDetails:
+                                referencedEntity.PostResolveConfigure((ComponentEntity e) =>
+                                {
+                                    e.PackageId = referencedEntities.GetByGlobal((GlobalAddress)objectDetails.PackageAddress).DatabaseId;
+                                    e.BlueprintName = objectDetails.BlueprintName;
+                                });
+                                break;
+                            default:
+                                throw new UnreachableException();
```

Comment:
can you add some message here with  `typeInfoSubstate.Details` to improove observability.

Reply:
Fixed